kombu/messaging.py

```diff
@@ -173,14 +173,43 @@ def publish(self, body, routing_key=None, delivery_mode=None,
                 # XXX declare should be a Set.
                 declare.append(self.exchange)
 
+        old_transport_opts = None
         if retry:
             _publish = self.connection.ensure(self, _publish, **retry_policy)
-        return _publish(
+
+            # If transport options are not provided for the connection
+            # override them with the retry policy provided for the publisher.
+            # This is done in order to workaround the fact that
+            # the default channel ensures that the connection exists
+            # to allow broker failover.
+            # This operation uses the retry policy of the connection instead
+            # of the publisher.
+            # See https://github.com/celery/kombu/commit/816e3dc.
+            old_transport_opts = self.connection.transport_options or {}
+            # Since we don't want these defaults to last we copy the dictionary
+            temp_transport_opts = self.connection.transport_options.copy()
+            temp_transport_opts.setdefault('max_retries',
+                                           retry_policy.get('max_retries'))
+            temp_transport_opts.setdefault('interval_start',
+                                           retry_policy.get('interval_start'))
+            temp_transport_opts.setdefault('interval_step',
+                                           retry_policy.get('interval_step'))
+            temp_transport_opts.setdefault('interval_max',
+                                           retry_policy.get('interval_max'))
+            self.connection.transport_options = temp_transport_opts
```

Comment:
How about transferring this operation to `Connection` and using a [context manager](https://docs.python.org/3.6/library/contextlib.html#contextlib.contextmanager)? It will be reusable too.

```python
with self.connection.set_temporary_options(interval_max=...):
    return _publish(...)
```

Reply:
I agree that's better.

Reply:
is this addressed?

Reply:
No I haven't had the time to deal with it.

Reply:
If you want me to push on your branch let me know